qutebrowser/browser/webengine/webview.py

```diff
@@ -238,3 +239,16 @@ def acceptNavigationRequest(self,
             is_main_frame=is_main_frame)
         self.navigation_request.emit(navigation)
         return navigation.accepted
+
+    def chooseFiles(
+        self,
+        mode: QWebEnginePage.FileSelectionMode,
+        old_files: Iterable[str],
+        accepted_mimetypes: Iterable[str],
+    ) -> List[str]:
+        """Override chooseFiles to (optionally) invoke custom file uploader."""
+        if config.val.fileselect.handler == "default":
+            return super().chooseFiles(mode, old_files, accepted_mimetypes)
+
```

Comment:
Perhaps add `assert handler == "external", handler` here, so this fails rather than doing the wrong thing if we ever add a new value there and forget to handle it here.

(At that point it probably makes sense again to do `handler = config.val.fileselect.handler` - sorry for that!

Reply:
Done